src/app/examples/grid-base-row-editing.component.ts

```diff
@@ -0,0 +1,265 @@
+
+import { Component, OnInit, ViewEncapsulation } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import { Subscription } from 'rxjs';
+
+import {
+  AngularGridInstance,
+  Column,
+  Editors,
+  FieldType,
+  Formatters,
+  GridOption,
+} from '../modules/angular-slickgrid';
+
+const NB_ITEMS = 20;
+
+@Component({
+  encapsulation: ViewEncapsulation.None,
+  styleUrls: ['./grid-base-row-editing.component.scss'],
+  templateUrl: './grid-base-row-editing.component.html'
+})
+export class GridBaseRowEditingComponent implements OnInit {
+  private subscriptions: Subscription[] = [];
+  angularGrid!: AngularGridInstance;
+  gridOptions!: GridOption;
+  columnDefinitions!: Column[];
+  dataset!: any[];
+  selectedLanguage = '';
+  selectedLanguageFile = '';
+  fetchResult = '';
+  statusClass = 'alert alert-light';
+  statusStyle = 'display: none';
+
+  constructor(private translate: TranslateService) {
+    // always start with English for Cypress E2E tests to be consistent
+    const defaultLang = 'en';
+    this.translate.use(defaultLang);
+    this.selectedLanguage = defaultLang;
+  }
+
+  ngOnInit() {
+    // mock some data (different in each dataset)
```

Comment:
this comment is a copy leftover from sample 1

Reply:
ah yeah didn't notice, I went rather quickly before going at the gym. thanks for the review